### PR TITLE
Ensure desktop dialogs' title bars are visible

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
In particular, the "Help" dialog shown for new users.

### CODE REVIEW

**Comment by ctrlaltdavid** [comment]
Testing
- No HMD connected.
- Start Interface.
- Settings > General > Desktop tablet becomes toolbar.
- Resize Interface window to small (not high enough to fit whole of Help dialog in window).
- Restart Interface.
- Open Help window.
- Help window's title bar with "x" button should be fully visible on-screen.
- Similarly for other dialogs with title bars such as Running Scripts.

If repeating tests:
- Delete C:\Users\USERNAME\AppData\Roaming\High Fidelity - PRNNNNN\Interface.json first.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/787/HighFidelity-Beta-PR10719-3cee2b3764adf598af412ce4664da15df9788409.exe

**Comment by ctrlaltdavid** [comment]
Test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/788/HighFidelity-Beta-PR10719-3cee2b3764adf598af412ce4664da15df9788409.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/788/HighFidelity-Beta-PR10719-3cee2b3764adf598af412ce4664da15df9788409.dmg

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

No HMD connected. **Check**
Start Interface.**Check**
Settings > General > Desktop tablet becomes toolbar.**Check**
Resize Interface window to small (not high enough to fit whole of Help dialog in window).**Check**
Restart Interface.**Check**
Open Help window.**Check**
Help window's title bar with "x" button should be fully visible on-screen. **Pass**
Similarly for other dialogs with title bars such as Running Scripts. **Pass**
Notes: Tested: Goto, People, Running Scripts, Market, Audio, Help, etc.

Please let me know if there is anything more I can do/provide to assist.

**Comment by tapalisa** [comment]
image
It should fix this situation as encountered by user testers

File: interface/resources/qml/desktop/Desktop.qml [hunk 1]
```diff
@@ -353,6 +353,14 @@ FocusScope {
         showDesktop();
     }
 
+    function ensureTitleBarVisible(targetWindow) {
+        // Reposition window to ensure that title bar is vertically inside window.
+        if (targetWindow.frame && targetWindow.frame.decoration) {
+            var topMargin = -targetWindow.frame.decoration.anchors.topMargin;  // Frame's topMargin is a negative value.
```
Comment (jherico) [inline interface/resources/qml/desktop/Desktop.qml]: this only accounts for the Y axis.  do we want to ensure that some or all of the X axis of the decoration is visible as well?

Comment (ctrlaltdavid) [inline interface/resources/qml/desktop/Desktop.qml]: @jherico The functions that this functions is called in - `centerOnVisible()` and 1repositionOnVisible()` ensure that the dialog is visible in the general sense (i.e., both x- and y-axes). This function just moves the result down if need be to get the title bar visible.

**Pull request merged**